sdk/schema/plugin.go

```diff
@@ -68,19 +68,13 @@ func (p Plugin) Validate() (bool, ValidationReport) {
 
 	report.AddCheck(ValidationCheck{
 		Description: "Has a credential type or executable defined",
-		Assertion:   len(p.Credentials) > 0 && len(p.Executables) > 0,
+		Assertion:   len(p.Credentials) > 0 || len(p.Executables) > 0,
 		Severity:    ValidationSeverityError,
 	})
 
 	report.AddCheck(ValidationCheck{
 		Description: "Has no more than one credential type defined. Plugins with multiple credential types are not supported yet",
-		Assertion:   len(p.Credentials) == 1,
-		Severity:    ValidationSeverityError,
-	})
-
-	report.AddCheck(ValidationCheck{
-		Description: "Credentials referenced in executables are included in the same plugin definition",
-		Assertion:   CredentialReferencesInCredentialList(p),
+		Assertion:   len(p.Credentials) <= 1,
 		Severity:    ValidationSeverityError,
 	})
```

Comment:
For my understanding: is there a specific reason this needs to go in this PR? Or is it just a matter of cleaning up an outdated rule?

Reply:
This will be needed in the future but for now we don't need to have it. I removed all validation checks which were not absolutely neccessary for this PR

Reply:
Actually, after I introduced this check back in, it was still failing. I now made some adjustments to it so that it adheres to the new credential usage definition.